### PR TITLE
allow setting offset for dav search queries not limited to the users home storage

### LINKED ISSUE #24417: S3 Objectstore as Primary Storage, Files API malfunction.
When using S3 objectstore as primary storage, apps such as the `Photos` app and the `Recommendations` app fail to show new files that have been added to the server.
Considering that this happens in not only a single app but multiple apps, which are all using the `Files API`, the problem is suspected to be a bug with the API itself, and not the apps. 

https://github.com/nextcloud/photos/issues/522
https://github.com/nextcloud/recommendations/issues/337

There is no errors within the browser, or logged error messages on the server side.
It can only be concluded, that the API itself is to fault.
Thanks!

### PR DESCRIPTION
**Pull request by icewind1991**
multi-storage offset is cheap (enough) now

Signed-off-by: Robin Appelman <robin@icewind.nl>

### CODE REVIEW

**Review by skjnldsv — APPROVED** [review]

File: apps/dav/lib/Files/FileSearchBackend.php [hunk 1]
```diff
@@ -275,7 +275,7 @@ private function transformQuery(Query $query): ISearchQuery {
 		// TODO offset
 		$limit = $query->limit;
 		$orders = array_map([$this, 'mapSearchOrder'], $query->orderBy);
-		$offset = 0;
+		$offset = $limit->firstResult;
```
Comment (juliushaertl) [inline apps/dav/lib/Files/FileSearchBackend.php:278]: I guess the todo from 3 lines above can be removed then.

**Review by juliusknorr — CHANGES_REQUESTED** [review]

File: apps/dav/lib/Files/FileSearchBackend.php (hunk 1 above)
Comment (icewind1991) [reply apps/dav/lib/Files/FileSearchBackend.php:277]: done

**Review by juliusknorr — APPROVED** [review]

**Comment by szaimen** [comment]
🎉

**Comment by szaimen** [comment]
Backport to stable21?

**Pull request merged**